### PR TITLE
Get rid of unsupported operating systems: Android 4.4 and less, IOS 9.3 and less.

### PR DESCRIPTION
**Pull request by jacksonrr3**
Removed outdated code due to the end of support Android less than 5 version and iOS less than 10 version.
Updated `mobile_ua` version at the mobile tests.

### CODE REVIEW

File: js/core/devices.js [hunk 1]
```diff
@@ -91,8 +91,7 @@ const uaParsers = {
         const isPhone = /mobile/i.test(userAgent);
         const matches = userAgent.match(/android (\d+)\.?(\d+)?\.?(\d+)?/i);
         const version = matches ? [parseInt(matches[1], 10), parseInt(matches[2] || 0, 10), parseInt(matches[3] || 0, 10)] : [];
-        const worseThan4_4 = version.length > 1 && (version[0] < 4 || version[0] === 4 && version[1] < 4);
-        const grade = worseThan4_4 ? 'B' : 'A';
+        const grade = 'A';
```
Comment (nightskylark) [inline js/core/devices.js]: Are we going to deprecate this field?

File: testing/tests/DevExpress.core/devices.tests.js [hunk 2]
```diff
@@ -50,21 +46,7 @@ QUnit.test('ios by userAgent', function(assert) {
...
-
-    device = fromUA(userAgents.android_4_4_0);
-
-    assert.equal(device.platform, 'android', 'platform is android');
-    assert.equal(device.version.toString(), '4,4,0', 'correct version');
-    assert.equal(device.deviceType, 'phone', 'deviceType is phone');
-    assert.equal(device.grade, 'A', 'grade is A');
-
-    device = fromUA(userAgents.android_tablet_7_1_1);
+    let device = fromUA(userAgents.android_tablet_7_1_1);
```
Comment (nightskylark) [inline testing/tests/DevExpress.core/devices.tests.js:49]:
```suggestion
    const device = fromUA(userAgents.android_tablet_7_1_1);
```

File: testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js [hunk 3]
```diff
@@ -1244,25 +1244,6 @@ QUnit.module('dateView integration', {
         assert.equal($element.dxDateBox('instance')._strategy.NAME, 'Native', 'correct strategy is chosen');
     });
 
-    QUnit.test('pickerType should be \'rollers\' on android < 4.4 (Q588373, Q588012)', function(assert) {
```
Comment (nightskylark) [inline testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js:1247]: Do we still need 'rollers'?

**Review by nightskylark — CHANGES_REQUESTED** [review]

File: testing/tests/DevExpress.core/devices.tests.js (hunk 2 above)
Comment (jacksonrr3) [reply testing/tests/DevExpress.core/devices.tests.js:49]: It need be `let`, cause we change this variable below.

Comment (nightskylark) [reply testing/tests/DevExpress.core/devices.tests.js:49]: Can you show me the line where the `device` variable is reassigned?

**Review by nightskylark — DISMISSED** [review]

**Review by Zedwag — DISMISSED** [review]

File: js/events/gesture/emitter.gesture.scroll.js [hunk 4]
```diff
@@ -154,10 +153,10 @@ let PointerLocker = TimeoutLocker.inherit((function() {
 })());
 
 (function() {
-    const ios8_greater = realDevice.ios && compareVersions(realDevice.version, [8]) >= 0;
-    const android5_greater = realDevice.android && compareVersions(realDevice.version, [5]) >= 0;
+    const iosDevice = realDevice.ios;
+    const androidDevice = realDevice.android;
```
Comment (DokaRus) [inline js/events/gesture/emitter.gesture.scroll.js]:
```suggestion
    const { ios: isIos, android: isAndroid } = realDevice;
```

File: js/ui/date_box/ui.date_box.mask.js [hunk 5]
```diff
@@ -134,7 +134,7 @@ const DateBoxMask = DateBoxBase.inherit({
 
     _useBeforeInputEvent: function() {
         const device = devices.real();
```
Comment (DokaRus) [inline js/ui/date_box/ui.date_box.mask.js]:
```suggestion
        const isAndroid = devices.real().android;
```

File: js/ui/date_box/ui.date_box.mask.js [hunk 6]
```diff
@@ -134,7 +134,7 @@ const DateBoxMask = DateBoxBase.inherit({
...
-        return device.android && device.version[0] > 4;
+        return device.android;
```
Comment (DokaRus) [inline js/ui/date_box/ui.date_box.mask.js]:
```suggestion
        return isAndroid;
```

**Review by DokaRus — APPROVED** [review]

**Pull request closed without merging**